app/decorators/swagger/route_decorator.rb

```diff
@@ -62,9 +63,71 @@ def query_parameters
     end
   end
 
+  def path_parameters
+    path_parameters_names.map do |parameter|
+      {
+          name: parameter,
+          in: 'path',
+          required: true,
+          schema: {
+              type: "string"
+          }
+      }
+    end
+  end
+
   def security
     return unless security_scheme
 
     [{security_scheme.key => []}]
   end
+
+  def x_amazon_apigateway_integration(api_gateway_integration)
+    return unless api_gateway_integration
+
+    request_parameters = {}
+    path_parameters_names.each do |parameter|
+      request_parameters['integration.request.path.' + parameter] = 'method.request.path.' + parameter
+    end
+
+    {
+      cacheKeyParameters: request_parameters.keys,
+      httpMethod: http_method,
+      passthroughBehavior: 'when_no_match',
+      requestParameters: request_parameters,
+      timeoutInMillis: api_gateway_integration.timeout_in_millis,
+      type: 'http_proxy',
+      uri: api_gateway_integration.uri_prefix + normalized_url(remove_plus_signs: true)
+    }
+  end
+
+  def path_parameters_names
+    parts = url.split('/')
+    parameter_parts = parts.select do |part|
+      part.starts_with?(':')
+    end
+    parameter_parts.map do |part|
+      if part.ends_with?('+')
+        part[1..-2]
+      else
+        part[1..-1]
+      end
+    end
+  end
+
+  def normalized_url(remove_plus_signs: false)
+    parts = url.split('/')
```

Comment:
Rather than rewrite the logic of of url params parsing maybe you can reuse it?

path_parameters_names.inject(url) do |url, parameter_name|
  pattern = remove_plus_signs ? /:#{parameter_name}\+?/ : /:#{parameter_name}/
  url.gsub(pattern, "{#{parameter_name}}")
end

Reply:
Thanks. I just adapted the logic to be conform with what I had previously.